Utility.cpp

```diff
@@ -1,108 +1,141 @@
     {
-        //dragons don't need defensive items
-    }  
+        // dragons don't need defensive items
+    }
 }
-void useHelpfulItem(Character*, Item* item)
+void useHelpfulItem(Character* character, Item* item)
 {
-    if( auto* ch = dynamic_cast<Dwarf*>(character) )
+    if (auto* ch = dynamic_cast<Dwarf*>(character))
     {
-        ch->boostHitPoints(item->getBoost() * 2);
+        ch->boostHitPoints(static_cast<int>(item->getBoost() * 2));
     }
-    else if( auto* ch = dynamic_cast<Paladin*>(character) )
+    else if (auto* ch = dynamic_cast<Paladin*>(character))
     {
-        ch->boostHitPoints(item->getBoost() * 1.5);
+        ch->boostHitPoints(static_cast<int>(item->getBoost() * 1.5));
     }
-    else if( auto* ch = dynamic_cast<DragonSlayer*>(character))
+    else if (auto* ch = dynamic_cast<DragonSlayer*>(character))
     {
-        ch->boostHitPoints(item->getBoost() * 1.25);
+        ch->boostHitPoints(static_cast<int>(item->getBoost() * 1.25));
     }
-    else if( auto* ch = dynamic_cast<Dragon*>(character) )
+    else if (auto* ch = dynamic_cast<Dragon*>(character))
     {
-        //dragons don't carry helpful items!
+        // dragons don't carry helpful items!
     }
 }
-void useAttackItem(Character*, Item* item)
+void useAttackItem(Character* character, Item* item)
 {
-    if( auto* ch = dynamic_cast<Dwarf*>(character) )
+    if (auto* ch = dynamic_cast<Dwarf*>(character))
     {
-        ch->boostAttackDamage(item->getBoost() * 1.5);
+        ch->boostAttackDamage(static_cast<int>(item->getBoost() * 1.5));
     }
-    else if( auto* ch = dynamic_cast<Paladin*>(character) )
+    else if (auto* ch = dynamic_cast<Paladin*>(character))
     {
-        ch->boostAttackDamage(item->getBoost() * 1.33);
+        ch->boostAttackDamage(static_cast<int>(item->getBoost() * 1.33));
     }
-    else if( auto* ch = dynamic_cast<DragonSlayer*>(character))
+    else if (auto* ch = dynamic_cast<DragonSlayer*>(character))
     {
-        assert(false);
-        //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
-        //so their attack item should boost their attack damage by a factor of 10
-        //this means you need to GET the attack damage, multiply it by the item's boost, and BOOST the attackDamage with that multiplied value.  
-        //check Character.h for available member functions you can use.
+        // DragonSlayers get a 10x boost when attacking dragons, from their
+        // attack item. so their attack item should boost their attack damage by
+        // a factor of 10 this means you need to GET the attack damage, multiply
+        // it by the item's boost, and BOOST the attackDamage with that
+        // multiplied value. check Character.h for available member functions
+        // you can use.
+        auto atkDamage = ch->getAttackDamage();
+        ch->boostAttackDamage(
+            static_cast<int>(atkDamage * item->getBoost() * 10));
```

Comment:
`item->getBoost()` should return 10. You shouldn't need to multiply the boost by anything. 
> their attack item should boost their attack damage by a factor of 10

- [ ] Fix this line according to the spec.